### PR TITLE
Add documentation on how the `shoot-rsyslog-relp` extension configuration is applied to shoots

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
This PR adds development documentation on how the `shoot-rsyslog-relp` extension resource is reconciled and how its configuration is applied to shoot clusters via the OperatingSystemConfig webhook that the extension serves.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by ialidzhikov — COMMENTED** [review]
Minor suggestions, otherwise lgtm.

File: docs/development/shoot-rsyslog-relp.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+1. As part of the Shoot reconciliation flow, gardenlet deploys the OperatingSystemConfig resource.
+1. The shoot-rsyslog-relp extension serves a webhook that mutates the OperatingSystemConfig resource for Shoots having the shoot-rsyslog-relp extension enabled (the corresponding namespace gets labeled by gardenlet with `extensions.gardener.cloud/shoot-rsyslog-relp=true`). [pkg/webhook/operatingsystemconfig/ensurer.go](../../pkg/webhook/operatingsystemconfig/ensurer.go) contains implementation of the [genericmutator.Ensurer](https://github.com/gardener/gardener/blob/v1.82.0/extensions/pkg/webhook/controlplane/genericmutator/mutator.go) interface.
+    1. The webhook renders the [60-audit.conf.tpl](../../pkg/webhook/operatingsystemconfig/resources/templates/scripts/configure-rsyslog.tpl.sh) template and appends it to the OperatingSystemConfig files. When rendering the template, the configuration of the shoot-rsyslog-relp extension is used to fill in the required template values. The file is eventually installed as `/var/lib/rsyslog-relp-configurator/rsyslog.d/60-audit.conf` on the host os.
```
[the remark this answers is not in the document]
Comment (plkokanov) [reply docs/development/shoot-rsyslog-relp.md:17]: Here and other places, "eventually" is used to mean once the files are downloaded by the node agent (or cloud config downloader).

[the remark this answers is not in the document]
Comment (plkokanov) [reply docs/development/shoot-rsyslog-relp.md:17]: I can remove it to avoid confusion.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**